From de29c3c1d57647a27639ffb9c06cbe4952f30a55 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Sun, 17 May 2020 15:27:46 +0300
Subject: [PATCH] [SW-5080] preboot: add build version with timestamps

buildversion-habanalabs contains the same version information as in
localversion-habanalabs with the addition of timestamp. This addition
will be used when verifying a drity firmware's version that was written
to the card.

Change-Id: I7bb73665b916a225d76371aba07e1872a16f4849
---

diff --git a/Makefile b/Makefile
index ff5b9e9..2d042ac 100644
--- a/Makefile
+++ b/Makefile
@@ -41,7 +41,7 @@
 PPBOOT_OBJECTS += $(PPBOOT_APP_C:.c=.o)
 AUTO_GENERATED_FILE := auto_generated.h
 
-all: ppboot.bin preboot.bin pcie_phy_fw.bin
+all: ppboot.bin preboot.bin pcie_phy_fw.bin buildversion-habanalabs
 
 %.bin: %
 	$(OBJCOPY) -O binary -R .note.gnu.build-id $< $@
@@ -56,6 +56,9 @@
 	@( printf '#define PREBOOT_VER "%s"\n' '$(shell cut -d- -f2- localversion-habanalabs)' ) > $@
 	@LANG=en_us date +'#define PREBOOT_DATE "%b %d %Y"' >> $@
 	@date +'#define PREBOOT_TIME "%T"' >> $@
+
+buildversion-habanalabs: include/preboot.h $(AUTO_GENERATED_FILE)
+	echo "VERSION_STR" | gcc -include $< $(CFLAGS) -P -E -o - - | tail -1 | sed 's/\" \+\"//g;s/["\n]//g' | cut -d' ' -f2- | tr -d '\n' > $@
 
 pcie_phy_fw.bin: $(HL_DEV)/pcie_phy_fw.i
 	@cut -d' ' -f2- $< | xxd -r -p | dd status=none bs=16 count=$$(($(FW_SIZE) / 16)) conv=sync,swab of=$@
@@ -73,6 +76,6 @@
 	@$(CC) $(SFLAGS) -o $@ $<
 
 clean:
-	rm -f $(PREBOOT_OBJECTS) $(AUTO_GENERATED_FILE)
+	rm -f $(PREBOOT_OBJECTS) $(AUTO_GENERATED_FILE) buildversion-habanalabs
 
 .PHONY: clean
